### PR TITLE
Pass through m4/gcc-like variables

### PR DESCRIPTION
**Pull request by Jenselme**
That's just a rebase of #6 from @jmmills 

Close #6, #5, #2

### CODE REVIEW

File: jinja2cli/cli.py [hunk 1]
```diff
@@ -188,11 +188,14 @@ def cli(opts, args):
...
         extensions.append(ext)
 
+    data.update(parse_kv_string(opts.D or []))
     output = render(template_path, data, extensions)
 
     sys.stdout.write(output)
     sys.exit(0)
 
+def parse_kv_string(pairs):
+    return dict(pair.split('=',1) for pair in pairs)
```
Comment (mattrobenolt) [inline jinja2cli/cli.py]: Space after comma plz.

Comment (Jenselme) [inline jinja2cli/cli.py]: Nice catch. PR rebased and updated.

**Comment by mattrobenolt** [comment]
At a glance this looks good, but it's super late here, so I'll review in the morning and likely merge then.

Thanks for the pull requests! :)

**Comment by mattrobenolt** [comment]
@Jenselme So one thing that's a bit odd with this is it conflicts with reading from stdin.

For example, if you do:

```
$ jinja2 -Dfoo=bar file.jinja2
```

It'll hang waiting for `stdin`, but we've supplied values.

Now, I can see this going two ways:

* Ignore `stdin` when `-D` is passed and treat `-D` as the data values only
* Merge `-D` in with any other values, and don't let `-D` work standalone. (current implementation)

Did you have an opinion?

**Comment by Jenselme** [comment]
I think it can be interesting to have `-D` work alone. In my opinion, the user is supplying values with `-D`, he/she won't expect to use stdin to provide them. So I would disable `stdin` if `-D` is present. Do you want me to write a patch for that?

**Comment by mattrobenolt** [comment]
Sure. If not, I'll get to it tonight.

**Comment by mattrobenolt** [comment]
Also, that was my hunch and that's what I attempted to do first. :)

**Comment by Jenselme** [comment]
> Sure. If not, I'll get to it tonight.

Go ahead, I won't be able to write it before on or two days.

**Comment by jmmills** [comment]
I'm sorry, I totally forgot about this PR (it was a year ago). Is there something that I need to fix about it?

**Comment by mattrobenolt** [comment]
@jmmills Nope. We're good here. :)

**Pull request merged**